htdocs/core/lib/files.lib.php

```diff
@@ -129,39 +129,40 @@ function dol_dir_list($path, $types = "all", $recursive = 0, $filter = "", $excl
 				}
 				// Check if file is qualified
 				foreach ($excludefilterarray as $filt) {
-					if (preg_match('/'.$filt.'/i', $file) || preg_match('/'.$filt.'/i', $fullpathfile)) {
+					if (preg_match('{'.$filt.'}i', $utf8_file) || preg_match('{'.$filt.'}i', $mixed_fullpathfile)) {
```

Comment:
The renamed variable makes it clear that the matching is not ok here / on the left side with match the regex with an utf-8 encoded filed, and on the write the directory is in native encoding while the basename is in utf8.